content-security-policy/script-src/script-src-strict_dynamic_meta_tag.html

```diff
@@ -0,0 +1,94 @@
+<!DOCTYPE HTML>
+<html>
+
+<head>
+    <title>A `strict-dynamic` policy can be served in a META tag.</title>
+    <meta http-equiv="Content-Security-Policy" content="script-src 'strict-dynamic' 'nonce-dummy'">
+    <script src='/resources/testharness.js' nonce='dummy'></script>
+    <script src='/resources/testharnessreport.js' nonce='dummy'></script>
+
+    <!-- CSP served: script-src 'strict-dynamic' 'nonce-dummy' -->
+</head>
+
+<body>
+    <h1>A `strict-dynamic` policy can be served in a META tag.</h1>
+    <div id='log'></div>
+
+    <script nonce='dummy'>
+    document.addEventListener('securitypolicyviolation', function(e) {
+        assert_unreached('No CSP violation report has fired.');
+    });
+
+    async_test(function(t) {
+        var e = document.createElement('script');
+        e.id = 'appendChild';
+        e.src = 'simpleSourcedScript.js';
+        e.onload = t.step_func_done(function() {
+            assert_equals(document.getElementById('appendChild').getAttribute('executed'), '1');
+        });
+        e.onerror = t.unreached_func('Error should not be triggered.');
+        document.body.appendChild(e);
+    }, 'Script injected via `appendChild` is allowed with `strict-dynamic`.');
+</script>
+
+<script nonce='dummy'>
+    async_test(function(t) {
+        var e = document.createElement('script');
+        e.id = 'appendChild-incorrectNonce';
+        e.src = 'simpleSourcedScript.js';
+        e.setAttribute('nonce', 'wrong');
+        e.onload = t.step_func_done(function() {
+            assert_equals(document.getElementById('appendChild-incorrectNonce').getAttribute('executed'), '1');
+        });
+        e.onerror = t.unreached_func('Error should not be triggered.');
+        document.body.appendChild(e);
+    }, 'Script injected via `appendChild` is allowed with `strict-dynamic`, even if it carries an incorrect nonce.');
+</script>
+
+<script nonce='dummy'>
+    async_test(function(t) {
+        var e = document.createElement('script');
+        e.id = 'appendChild-textContent';
+        e.setAttribute('executed', '0');
+        e.textContent = "document.currentScript.setAttribute('executed', '1');";
+        e.onerror = t.unreached_func('Error should not be triggered.');
+        var observer = new MutationObserver(function(mutations) {
```

Comment:
This is a bit complicated. Did you consider just putting a test step function onto the global that the script could call? Like:

```
var e = document.createScript('script');
...
window.appendChildViaTextContent = t.step_func_done();
e.textContent = "appendChildViaTextContent();"
```

That leaves one global dangling around for each test, but seems more straightforward.